Assets/Scripts/UIManager.cs

```diff
@@ -23,17 +23,34 @@ public class UIManager : MonoBehaviour
     [SerializeField]
     private Button _restartLevelButton;
 
+    [SerializeField]
+    private Button _restartButton;
+
+    [SerializeField]
+    private Button _premadeLevelButton;
+
+    [SerializeField]
+    private Image _optionsMenuBackground;
+
+    [SerializeField]
+    private ShapesManager _shapesManager;
+
+    private bool _isOptionsMenuOpen = false;
```

Comment:
is there a reason you decided to keep this and not use the onClick event in the unity handler to pass the value to you?

Reply:
Currently, I haven't been able to get the OnClick event to work properly.

Reply:
oh that's strange, i wonder why that is?